flink-formats/flink-orc/src/main/java/org/apache/flink/orc/OrcFilters.java

```diff
@@ -0,0 +1,273 @@
+		return new OrcSplitReader.Equals(colName, litType, literal);
+	}
+
+	private static OrcSplitReader.Predicate convertNotEquals(String colName, PredicateLeaf.Type litType, Serializable literal) {
+		return new OrcSplitReader.Not(convertEquals(colName, litType, literal));
+	}
+
+	private static OrcSplitReader.Predicate convertGreaterThan(String colName, PredicateLeaf.Type litType, Serializable literal) {
+		return new OrcSplitReader.Not(
+				new OrcSplitReader.LessThanEquals(colName, litType, literal));
+	}
+
+	private static OrcSplitReader.Predicate convertGreaterThanEquals(String colName, PredicateLeaf.Type litType, Serializable literal) {
+		return new OrcSplitReader.Not(
+				new OrcSplitReader.LessThan(colName, litType, literal));
+	}
+
+	private static OrcSplitReader.Predicate convertLessThan(String colName, PredicateLeaf.Type litType, Serializable literal) {
+		return new OrcSplitReader.LessThan(colName, litType, literal);
+	}
+
+	private static OrcSplitReader.Predicate convertLessThanEquals(String colName, PredicateLeaf.Type litType, Serializable literal) {
+		return new OrcSplitReader.LessThanEquals(colName, litType, literal);
+	}
+
+	public static OrcSplitReader.Predicate toOrcPredicate(Expression expression) {
+		if (expression instanceof CallExpression) {
+			CallExpression callExp = (CallExpression) expression;
+			if (FILTERS.get(callExp.getFunctionDefinition()) == null) {
+				// unsupported predicate
+				LOG.debug("Unsupported predicate [{}] cannot be pushed into OrcFileSystemFormatFactory.", expression);
+				return null;
+			}
+			return FILTERS.get(callExp.getFunctionDefinition()).apply(callExp);
+		} else {
+			// unsupported predicate
+			LOG.debug("Unsupported predicate [{}] cannot be pushed into OrcFileSystemFormatFactory.", expression);
+			return null;
+		}
+	}
+
+	private static String getColumnName(CallExpression comp) {
+		if (literalOnRight(comp)) {
+			return ((FieldReferenceExpression) comp.getChildren().get(0)).getName();
+		} else {
+			return ((FieldReferenceExpression) comp.getChildren().get(1)).getName();
+		}
+	}
+
+	private static boolean literalOnRight(CallExpression comp) {
+		if (comp.getChildren().size() == 1 && comp.getChildren().get(0) instanceof FieldReferenceExpression) {
+			return true;
+		} else if (isLit(comp.getChildren().get(0)) && isRef(comp.getChildren().get(1))) {
+			return false;
+		} else if (isRef(comp.getChildren().get(0)) && isLit(comp.getChildren().get(1))) {
+			return true;
+		} else {
+			throw new RuntimeException("Invalid binary comparison.");
+		}
+	}
+
+	private static PredicateLeaf.Type getLiteralType(CallExpression comp) {
+		if (literalOnRight(comp)) {
+			return toOrcType(((ValueLiteralExpression) comp.getChildren().get(1)).getOutputDataType());
+		} else {
+			return toOrcType(((ValueLiteralExpression) comp.getChildren().get(0)).getOutputDataType());
+		}
+	}
+
+	private static Optional<?> getLiteral(CallExpression comp) {
```

Comment:
I have some concern about literal structure..
Especially timestamp and date, Flink is `LocalTimestamp`, and `LocalDate`, I don't know what is ORC. You verify this in ITCase.

Reply:
Flink LocalDate -> ORC java.sql.Date

Reply:
By looking at the orc source code, orc is java.sql.Date and java.sql.Timestamp
The relevant code is in RecordReaderImpl.getBaseObjectForComparison()

Reply:
Maybe we can change the getLiteral() function like this:
```
private static Object getLiteral(CallExpression comp) {
        Object literalObj;
        if (literalOnRight(comp)) {
                ValueLiteralExpression valueLiteralExpression = (ValueLiteralExpression) comp.getChildren().get(1);
                literalObj = valueLiteralExpression.getValueAs(valueLiteralExpression.getOutputDataType().getConversionClass()).get();
        } else {
                ValueLiteralExpression valueLiteralExpression = (ValueLiteralExpression) comp.getChildren().get(0);
                literalObj = valueLiteralExpression.getValueAs(valueLiteralExpression.getOutputDataType().getConversionClass()).get();
        }

        if(literalObj instanceof LocalDate){
                LocalDate localDate = (LocalDate) literalObj;
                return Date.valueOf(localDate);
        }else if(literalObj instanceof LocalDateTime){
                LocalDateTime localDateTime = (LocalDateTime) literalObj;
                return Timestamp.valueOf(localDateTime);
        }else{
                return literalObj;
        }
}
```

Test sql in ITCase:

```
create table orcFilterTable (
        x string,
        y int,
        a int,
        b bigint,
        c boolean,
        d string,
        e decimal(8,4),
        f date,
        g timestamp
) with (
        'connector' = 'filesystem',
        'path' = '/path/to/data',
        'format' = 'orc'
)

// test date type
 select x, y from orcFilterTable where f = date '2020-01-01'

// test timestamp type
 select x, y from orcFilterTable where g = timestamp '2020-01-01 00:00:00'
```

Reply:
@JingsongLi  Do you have a better solution ?

Reply:
I think it is good, minor is we take a look to type first: `public Serialiable toOrcObject() {switch(typeRoot) case TIMESTAMP}...`.